lib/services/access_request.go

```diff
@@ -496,22 +496,27 @@ func UnmarshalAccessRequest(data []byte, opts ...MarshalOption) (AccessRequest,
 }
 
 // MarshalAccessRequest marshals the AccessRequest resource to JSON.
-func MarshalAccessRequest(req AccessRequest, opts ...MarshalOption) ([]byte, error) {
+func MarshalAccessRequest(r AccessRequest, opts ...MarshalOption) ([]byte, error) {
 	cfg, err := CollectOptions(opts)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	switch r := req.(type) {
+
+	version := r.GetVersion()
+	switch accessRequest := r.(type) {
 	case *AccessRequestV3:
+		if version != V3 {
```

Comment:
Is this meant as protection for manually crafted resource values?

Reply:
Yes, because it is possible to provide an invalid version, which will break when it comes time to unmarshal from the backend.